src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -1135,7 +1135,7 @@ public struct OperationAnalysisContext
     {
         private readonly IOperation _operation;
         private readonly ISymbol _containingSymbol;
-        private readonly Compilation _compilation;
+        private readonly SemanticModel _semanticModel;
```

Comment:
Please make similar changes to [OperationBlockAnalysisContext](http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs,f1dcb115dd76832e,references) and [OperationBlockStartAnalysisContext](http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs,1fa359e1b364953c,references)

Reply:
> Please make similar changes to OperationBlockAnalysisContext and OperationBlockStartAnalysisContext

I guess it would be good to understand what is the end goal for this change. For example, if we also need to make similar changes to OperationBlockAnalysisContext and OperationBlockStartAnalysisContext, then perhaps we should consider adding a helper method for an IOperation to extract SemanticModel that created it, or even add it to IOperation interface. Then, all clients can benefit from it everywhere.

---
In reply to: [182548021](https://github.com/dotnet/roslyn/pull/26187#discussion_r182548021) [](ancestors = 182548021)

Reply:
@JohanLarsson wants to use it here: https://github.com/dotnet/roslyn/pull/26172

specifically, right now he needs to do:

```c#
+                foreach (var symbol in context.Compilation.GetSemanticModel(literal.Syntax.SyntaxTree)
+                                                .LookupSymbols(literal.Syntax.SpanStart, name: value))
```

Having the semantic model available makes sense as not all information is available through IOperation.

Reply:
Basically, if an analyzer wants to use any of hte "query" operations on SemanticModel while handling some IOperation, there's no efficient way to do that.